### PR TITLE
[MM-42479] Reduce number of calls to get the subscription data

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
Prior to this change, the webapp would request the subscription data from the backend (ie, from CWS/Stripe) 3 times every refresh. This was because each of the announcement bars individually wrapped themselves with the withGetSubscription HOC - causing a race to happen, and 3 requests to be made. 

This PR wraps the announcement bar controller HOC in the withGetSubscriptionStats HOC, so that it's only done once. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42479

#### Related Pull Requests

- Has server changes (please link here)
- Has mobile changes (please link here)

#### Screenshots
Both of these screenshots are of the network tab in the chrome dev console, filtered for /api/v4/cloud
Before:
image
After:
image

#### Release Note

```release-note
None
```

### CODE REVIEW

**Comment by nickmisasi** [comment]
@neallred @pvev this is my first webapp contribution in a long time - my React is quite rusty. I _think_ this is a safe change, but please let me know if you have concerns. Removing the cloud license makes this HOC just not do anything, so it seems all good.

File: components/announcement_bar/cloud_announcement_bar/user_limit_announcement_bar.tsx [hunk 1]
```diff
@@ -174,4 +174,4 @@ class UserLimitAnnouncementBar extends React.PureComponent<Props> {
     }
 }
 
-export default withGetCloudSubscription(UserLimitAnnouncementBar);
+export default UserLimitAnnouncementBar;
```
Comment (pvev) [inline components/announcement_bar/cloud_announcement_bar/user_limit_announcement_bar.tsx:177]: I think you should remove the import statement for `withGetCloudSubscription` too.

File: components/announcement_bar/cloud_trial_announcement_bar/cloud_trial_announcement_bar.tsx [hunk 2]
```diff
@@ -205,4 +205,4 @@ class CloudTrialAnnouncementBar extends React.PureComponent<Props> {
     }
 }
 
-export default withGetCloudSubscription(CloudTrialAnnouncementBar);
```
Comment (pvev) [inline components/announcement_bar/cloud_trial_announcement_bar/cloud_trial_announcement_bar.tsx:208]: I think you should remove the import statement for `withGetCloudSubscription` too.

File: components/announcement_bar/payment_announcement_bar/payment_announcement_bar.tsx [hunk 3]
```diff
@@ -93,4 +93,4 @@ class PaymentAnnouncementBar extends React.PureComponent<Props> {
     }
 }
 
-export default withGetCloudSubscription(PaymentAnnouncementBar);
```
Comment (pvev) [inline components/announcement_bar/payment_announcement_bar/payment_announcement_bar.tsx:96]: I think you should remove the import statement for `withGetCloudSubscription` too.

File: components/announcement_bar/announcement_bar_controller.tsx [hunk 4]
```diff
@@ -19,16 +20,22 @@ type Props = {
     license?: ClientLicense;
     config?: Partial<ClientConfig>;
     canViewSystemErrors: boolean;
+    isCloud: boolean;
```
Comment (pvev) [inline components/announcement_bar/announcement_bar_controller.tsx:23]: not sure if I understand why are these values needed?

File: components/announcement_bar/index.ts [hunk 5]
```diff
@@ -11,6 +11,8 @@ import {getDisplayableErrors} from 'mattermost-redux/selectors/errors';
 import {dismissError} from 'mattermost-redux/actions/errors';
 import {getStandardAnalytics} from 'mattermost-redux/actions/admin';
 import {GenericAction} from 'mattermost-redux/types/actions';
+import {getCloudSubscription, getCloudCustomer, getSubscriptionStats} from 'mattermost-redux/actions/cloud';
```
Comment (neallred) [inline components/announcement_bar/index.ts:14]: I like the approach of moving all the calls and `withGet` wrapping one level up from the child bars to the parent controller. As part of that, I think the HOC should start making use of getCloudCustomer similar to how `payment_announcement_bar` componentDidMount does it, and remove the current payment_announcement_bar call.

File: components/announcement_bar/announcement_bar_controller.tsx (hunk 4 above)
Comment (nickmisasi) [reply components/announcement_bar/announcement_bar_controller.tsx:23]: Provides the necessary props to the with_get_cloud_subscription component

File: components/announcement_bar/index.ts (hunk 5 above)
Comment (nickmisasi) [reply components/announcement_bar/index.ts:14]: I see what you're saying - I think the reason this is like this is that the `payment_announcement_bar` is the only bar that actually utilizes the customer data - everything else is subscription only. I think this is a bit out of scope for this small change - you ok if I open a ticket for it and we can look into it later?

Comment (neallred) [reply components/announcement_bar/index.ts:14]: oh, sure.

**Review by neallred — APPROVED** [review]
LGTM, thanks @nickmisasi :ship: !

File: components/announcement_bar/index.ts (hunk 5 above)
Comment (nickmisasi) [reply components/announcement_bar/index.ts:14]: Here's that ticket: https://mattermost.atlassian.net/browse/MM-42602

File: components/announcement_bar/announcement_bar_controller.tsx (hunk 4 above)
Comment (pvev) [reply components/announcement_bar/announcement_bar_controller.tsx:23]: ok, got it. Thanks!

**Review by pvev — APPROVED** [review]
LGTM! thanks @nickmisasi :shipit:

**Pull request merged**